applications/finetune-quora-embeddings/main.py

```diff
@@ -0,0 +1,170 @@
+from modal import Image, gpu, Stub, Volume, Secret
+import os
+
+# Stub Configuration
+MODEL_ID = "BAAI/bge-base-en-v1.5"
+GPU_CONFIG = gpu.A100()
+
+# Dataset Configuration
+DATASET_NAME = "quora"
+DATASET_CONFIG = "default"
+DATASET_SPLIT = "train"
+DATASET_DIR = "/data"
+DATASET_VOLUME = Volume.persisted("datasets")
+
+# Embedding Configuration
+INCLUDE_DENSE_LAYER = True
+DENSE_OUT_FEATURES = 768
+
+# Finetune Configuration
+TEST_PERCENTAGE = 0.9
+BATCH_SIZE = 32
+EPOCHS = 3
+SCHEDULER = "warmuplinear"
+CHECKPOINT_VOLUME = Volume.persisted("checkpoints")
+CHECKPOINT_DIR = "/checkpoints"
+SAVE_CHECKPOINT = True
+MAX_CHECKPOINTS_SAVED = 3
+
+# Wandb Configuration
+WANDB_PROJECT_NAME = "quora-finetuning"
+
+stub = Stub("finetune-quora-embeddings")
+
+
+def download_model():
+    from sentence_transformers import SentenceTransformer
+
+    SentenceTransformer(MODEL_ID)
+
+
+requirements_txt = os.path.join(os.path.dirname(__file__), "requirements.txt")
```

Comment:
Tried highlighting the use of a `requirements.txt` file since I didn't see anyone using it else where... it was only present in a test in their repo